src/test/java/com/yelp/nrtsearch/server/grpc/ReplicationServerTest.java

```diff
@@ -492,4 +492,18 @@ public static void validateSearchResults(SearchResponse searchResponse) {
     SearchResponse.Hit secondHit = searchResponse.getHits(1);
     LuceneServerTest.checkHits(secondHit);
   }
+
+  @Test
+  public void testAddDocumentsOnReplicaFailure() throws IOException, InterruptedException {
+    // startIndex primary
+    GrpcServer.TestServer testServerPrimary =
+        new GrpcServer.TestServer(luceneServerPrimary, true, Mode.PRIMARY);
+
+    // startIndex replica
+    GrpcServer.TestServer testServerReplica =
+        new GrpcServer.TestServer(luceneServerSecondary, true, Mode.REPLICA);
+    testServerReplica.addDocuments();
+    assertEquals(testServerReplica.error, true);
```

Comment:
It would be good to verify that the error is as expected. You could modify the `GrpcServer` to also keep the `Throwable` that caused the error. That way you can verify the type and message.